src/Grpc.Net.Client/Balancer/Subchannel.cs

```diff
@@ -233,7 +233,21 @@ public void RequestConnection()
             }
         }
 
-        _ = ConnectTransportAsync();
+        // Don't capture the current ExecutionContext and its AsyncLocals onto the connect
+        bool restoreFlow = false;
+        if (!ExecutionContext.IsFlowSuppressed())
+        {
+            ExecutionContext.SuppressFlow();
+            restoreFlow = true;
+        }
+
+        _ = Task.Run(ConnectTransportAsync);
```

Comment:
An alternative would be to put an `await Task.Yield()` at the top of `ConnectTransportAsync`, or to leave it as-is.
This is fine, too.